lib/repl.js

```diff
@@ -1113,9 +1113,12 @@ REPLServer.prototype.defineCommand = function(keyword, cmd) {
   this.commands[keyword] = cmd;
 };
 
-REPLServer.prototype.memory = function memory(cmd) {
-  var self = this;
+REPLServer.prototype.memory = util.deprecate(
+  function(cmd) { _memory(this, cmd); },
```

Comment:
Nit: I'd use `_memory` directly to remove yet another wrapper.

Reply:
If I deprecate like so

```js
REPLServer.prototype.memory = util.deprecate(
  _memory,
  'REPLServer.memory() is deprecated',
  'DEP00XX');
```

then I will need to set `self = this` in `_memory`. E.g.

```js
function _memory(cmd) {
  self = this;
  self.lines = self.lines || [];
  self.lines.level = self.lines.level || [];
  // ... etc
}
```

However, in order for `this` to work correctly, I would need to modify all of the direct [internal calls to `_memory(self, cmd)`](https://github.com/nodejs/node/pull/16242/files#diff-b13d72249263845d8e8341db0426f9d3R496) so that they instead call `self.memory(cmd)`, which goes through the deprecation wrapper. IMO, I think this is a less optimal approach. 

With the current implementation, all internal calls go directly to `_memory()`, bypassing the wrapper altogether. Only those calls to `REPLServer.prototype.memory()` will hit it, and the idea is for this to go away anyway.